config/src/main/java/org/springframework/security/config/annotation/web/configurers/ExpressionUrlAuthorizationConfigurer.java

```diff
@@ -188,6 +190,14 @@ final ExpressionBasedFilterInvocationSecurityMetadataSource createMetadataSource
 			if (trustResolver != null) {
 				defaultHandler.setTrustResolver(trustResolver);
 			}
+			ApplicationContext context = http.getSharedObject(ApplicationContext.class);
+			if(context != null) {
```

Comment:
Maybe I'm missing something? I see that the context is required for `WebSecurityConfigurerAdapter` as the default for `Autowired.required` is true:

```
	@Autowired
	public void setApplicationContext(ApplicationContext context) {
		this.context = context;
	}

```